src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const stateClone = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(stateClone, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyRemove of keysToRemove) {
+          if (stateClone.hasOwnProperty(keyRemove)) {
```

Comment:
there is no need to verify if object contains it
when we try to delete non-existing key-value pair, it won't cause anything bad